trace/tests/system/flask/flask_system_test.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2017, OpenCensus Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import random
+import requests
+import shlex
+import signal
+import subprocess
+import time
+import uuid
+
+import unittest
+
+PROJECT = os.environ.get('GCLOUD_PROJECT_PYTHON')
+
+
+def wait_app_to_start():
+    """Wait the application to start running."""
+    cmd = 'until nc -z -v -w30 127.0.0.1 8080; do sleep 2; done'
+    os.system(cmd)
+
+
+def generate_header():
+    """Generate a trace header."""
+    trace_id = uuid.uuid4().hex
+    span_id = random.getrandbits(64)
+    trace_option = 1
+
+    header = '{}/{};o={}'.format(trace_id, span_id, trace_option)
+
+    return trace_id, span_id, header
+
+
+def run_application():
+    """Start running the flask application."""
+    cmd = 'python tests/system/flask/main.py'
+    process = subprocess.Popen(
+        cmd,
+        stdout=subprocess.PIPE,
+        shell=True,
+        preexec_fn=os.setsid)
+    return process
+
+
+class TestFlaskTrace(unittest.TestCase):
```

Comment:
If we could separate the "test harness" from the "start flask application", we could reduce some of the duplicated code between the django and flask tests.

Reply:
Yes, will fix in separate PR.